File: week01/day07/frontend/src/types/worker.ts
export interface Worker {
  id: number;
  user_id: number;
  name: string;
  phone: string;
  id_card: string;
  avatar: string | null;
  status: 'active' | 'inactive' | 'deleted';
  created_at: string;
}

export interface WorkerFormData {
  name: string;
  phone: string;
  id_card: string;
  avatar?: string;
}

File: week01/day07/backend/schemas/worker.py
from pydantic import BaseModel, Field


class WorkerCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=50)
    phone: str = Field(..., min_length=1, max_length=20)
    id_card: str = Field(..., min_length=1, max_length=18)
    avatar: str | None = None


class WorkerUpdate(BaseModel):
    name: str = Field(..., min_length=1, max_length=50)
    phone: str = Field(..., min_length=1, max_length=20)
    avatar: str | None = None


class WorkerStatusUpdate(BaseModel):
    status: str = Field(..., pattern="^(active|inactive)$")


class WorkerOut(BaseModel):
    id: int
    user_id: int
    name: str
    phone: str
    id_card: str
    avatar: str | None = None
    status: str
    created_at: str

    model_config = {"from_attributes": True}
